### PR TITLE
click_button() now accepts ids.

### PR DESCRIPTION
**Pull request by oalders**

### CODE REVIEW

File: lib/WWW/Mechanize.pm [hunk 1]
```diff
@@ -1850,6 +1854,11 @@ sub click_button {
     if ( $args{name} ) {
         $request = $form->click( $args{name}, $args{x}, $args{y} );
     }
+    elsif ( $args{id} ) {
```
Comment (ilmari) [inline lib/WWW/Mechanize.pm]: This fails for buttons with `id="0"`, which is allowed in HTML5.

File: t/local/click_button.t [hunk 2]
```diff
@@ -23,26 +25,34 @@ ok( $mech->is_html, 'Local page is HTML' );
...
-    $mech->click_button(number => 1);
+subtest 'click by id' => sub {
+    $mech->click_button(id => 'my-submit-id');
+    test_click( $mech );
 
-    like( $mech->uri, qr/formsubmit/, 'Clicking on button by number' );
-    like( $mech->uri, qr/submit=Go/,  'Correct button was pressed' );
-    like( $mech->uri, qr/cat_foo/,    'Parameters got transmitted OK' );
-    $mech->back;
+    ok(! eval { $mech->click_button(number => 2); 1 }, 'Button number out of range');
```
Comment (ilmari) [inline t/local/click_button.t]: This seems to be a copy-pasto from the number subtest.

File: lib/WWW/Mechanize.pm (hunk 1 above)
Comment (oalders) [inline lib/WWW/Mechanize.pm]: Nice catch!  Fixed in 53c37d98f1

File: t/local/click_button.t (hunk 2 above)
Comment (oalders) [inline t/local/click_button.t]: Oops!  Fixed in 3313ad8c7e

**Pull request merged**